### PR TITLE
Fix now in millis for ESQL search contexts

### LINKED ISSUE #103455: ESQL: `"lt[e]": "now"` in a range query `filter` erroneously returns nothing
### Description

This works:
```
POST _query
{
  "query": "from kibana_sample_data_logs | limit 10000",
  "filter": {
    "range": {
      "@timestamp": {
        "format": "strict_date_optional_time",
        "gte": "2023-12-12T13:00:00.000Z",
        "lt": "2023-12-12T14:00:00.000Z"
      }
    }
  }
}
```

But this returns empty values array:
```
POST _query
{
  "query": "from kibana_sample_data_logs | limit 10000",
  "filter": {
    "range": {
      "@timestamp": {
        "format": "strict_date_optional_time",
        "gte": "2023-12-12T13:00:00.000Z",
        "lt": "now"
      }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by dnhatn**
Currently, ESQL search contexts do not have the correct actual 'now' in milliseconds. Ideally, we should use a consistent 'now' in millis for all nodes participating in the execution of a search request. I will follow up with a PR to deserialize that value between nodes for version 8.13 and later.

Closes #103455

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlConfiguration.java [hunk 1]
```diff
@@ -112,6 +112,15 @@ public String query() {
...
     }
 
+    /**
+     * Returns the current time in milliseconds from the time epoch for the execution of this request.
+     * It ensures consistency by using the same value on all nodes involved in the search request.
+     * Note: Currently, it returns {@link System#currentTimeMillis()}, but this value will be serialized between nodes.
+     */
+    public long absoluteStartedTimeInMillis() {
+        return System.currentTimeMillis();
+    }
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlConfiguration.java:122]: Can we not use the inherited `now()` instead?

**Review by bpintea — APPROVED** [review]
Left one comment, it otherwise LGTM.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlConfiguration.java (hunk 1 above)
Comment (dnhatn) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlConfiguration.java:122]: It seems that `now` will be adjusted with the timezone?

**Comment by dnhatn** [comment]
Thanks @bpintea.

**Pull request merged**